### PR TITLE
Clarify what we mean by decoupled weight decay in the *AdamWs

### PR DESCRIPTION
**Pull request by janeyx99**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #144101

### CODE REVIEW

**Comment by janeyx99** [comment]
@ad8e Do you think this is clearer if we specify what we mean by decoupled weight decay? Resolving the rest of https://github.com/pytorch/pytorch/issues/104899 will take longer, but I'm hoping this can remove potential confusion.

**Comment by ad8e** [comment]
AdamW was decoupled in both senses, so "as in AdamW" could refer to both things. When reading "From the gradient-based update", it takes me some thinking to figure out the intended meaning, which is tied to the algorithm's technical details. Although it's technically correct.

"Do not accumulate weight decay in the momentum buffers" is a clearer description - and is also the core of why AdamW works, because weight decay doesn't make sense in the second moment.

Overall, it's probably not a big issue. A few people might have to think a bit more.

**Comment by ad8e** [comment]
Your new wording is better than mine, LGTM

**Review by albanD — APPROVED** [review]
Ok!
